tests/network/kea_server.pm

```diff
@@ -0,0 +1,62 @@
+# SUSE's openQA tests
+#
+# Copyright SUSE LLC
+# SPDX-License-Identifier: FSFAP
+
+# Package: kea
+# Summary: Basic kea test
+# - configure Kea dhcp4 with predefined config file
+# - start the Kea dhcp4 server and validate the config file
+# - enable DHCP on the client and reload network settings
+# - verify client assigned IP is within the pool
+# - verify client assigned IP changed from pre-set IP
+# - check the Kea lease file
+# - check the Kea logs for full DHCP handshake
+# Maintainer: QE Core <qe-core@suse.de>
+
+use base 'consoletest';
+use strict;
+use warnings;
+use testapi;
+use lockapi;
+use utils qw(zypper_call systemctl);
+use network_utils 'iface';
+use serial_terminal 'select_serial_terminal';
+
+sub run {
+    select_serial_terminal;
+
+    barrier_create('kea_setup', 2);
+    barrier_create('kea_finished', 2);
+    mutex_create('barrier_setup_done');
```

Comment:
1. the names of mutexes and barriers are IMO mess
2. Why you create barrier `kea_setup` when it's directly followed on line 44-45 by barrier `kea_finished` ? It's not needed as you have mutex `barrier_setup_done` which name is also false as it's before setup

Reply:
Sorry, the mutex is about barrier setup so ok, name is "fine". Just `kea_setup` does not make sense.

Reply:
And the mutex_create has to be before barriers, to make sure barriers will be created